### DISCUSSION TITLE
Possible license incompatibility (Nordic and GPL)

### DISCUSSION THREAD

**Discussion by fgaz** [meshtastic/firmware#4864] [General]
This project contains both code under GPL 3.0 and code under Nordic 5-Clause.

GPL 3.0 states:

https://github.com/meshtastic/firmware/blob/10869ea10a8954d2471ae120aa0e4c76327c4fe7/LICENSE#L210-L212

https://github.com/meshtastic/firmware/blob/10869ea10a8954d2471ae120aa0e4c76327c4fe7/LICENSE#L222-L228

and

https://github.com/meshtastic/firmware/blob/10869ea10a8954d2471ae120aa0e4c76327c4fe7/LICENSE#L388-L396

and

https://github.com/meshtastic/firmware/blob/10869ea10a8954d2471ae120aa0e4c76327c4fe7/LICENSE#L463-L469

Nordic 5-Clause states:

https://github.com/meshtastic/firmware/blob/10869ea10a8954d2471ae120aa0e4c76327c4fe7/src/platform/nrf52/softdevice/ble.h#L21-L25

To me these terms seem incompatible, meaning that nobody but Meshtastic LLC (the sole copyright holder due to the CLA) is allowed to legally distribute the Meshtastic firmware for nrf52 boards. Am I missing something?

**Comment by jp-bennett** [comment]
Good catch. I know at least some other projects don't consider this a problem, but we'll look into it.

**Comment by caveman99** [comment]
I talked to Thomas of Nordic Semiconductor about this and he also does not consider it a problem. These are **header files** we place there for CI building convenience to support different softdevices woithout needing to patch the toolchain. They're usually part of the nordic nRF5 sdk/nrfxlib which we link to anyway.
